### PR TITLE
[13.0]Multi company support for Scrap reason

### PR DESCRIPTION
**Pull request by bizzappdev**
added support for multi-company for scrap reason.

### CODE REVIEW

File: scrap_reason_code/views/stock_scrap_views.xml [hunk 1]
```diff
@@ -7,7 +7,11 @@
         <field name="inherit_id" ref="stock.stock_scrap_form_view" />
         <field name="arch" type="xml">
             <xpath expr="//group/group" position='inside'>
-                <field name="reason_code_id" />
+                <field
+                    name="reason_code_id"
+                    required="1"
+                    domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
```
Comment (rousseldenis) [inline scrap_reason_code/views/stock_scrap_views.xml:12]: IMHO you should write ir.rule for that. So, no need of domain

Comment (etobella) [reply scrap_reason_code/views/stock_scrap_views.xml:12]: Domain should be welcome, as on 13.0 we can work on several companies at the same time

Comment (rousseldenis) [reply scrap_reason_code/views/stock_scrap_views.xml:12]: Indeed, ok!

File: scrap_reason_code/views/stock_scrap_views.xml [hunk 2]
```diff
@@ -7,7 +7,11 @@
         <field name="inherit_id" ref="stock.stock_scrap_form_view" />
         <field name="arch" type="xml">
             <xpath expr="//group/group" position='inside'>
-                <field name="reason_code_id" />
+                <field
+                    name="reason_code_id"
+                    required="1"
```
Comment (bodedra) [inline scrap_reason_code/views/stock_scrap_views.xml]:
```suggestion
```

File: scrap_reason_code/views/stock_scrap_views.xml [hunk 3]
```diff
@@ -17,7 +21,11 @@
         <field name="inherit_id" ref="stock.stock_scrap_tree_view" />
         <field name="arch" type="xml">
             <xpath expr="//field[@name='product_uom_id']" position='after'>
-                <field name="reason_code_id" />
+                <field
+                    name="reason_code_id"
+                    required="1"
```
Comment (bodedra) [inline scrap_reason_code/views/stock_scrap_views.xml]:
```suggestion
```

File: scrap_reason_code/views/stock_scrap_views.xml [hunk 4]
```diff
@@ -27,7 +35,11 @@
         <field name="inherit_id" ref="stock.stock_scrap_form_view2" />
         <field name="arch" type="xml">
             <xpath expr="//group/group" position='inside'>
-                <field name="reason_code_id" />
+                <field
+                    name="reason_code_id"
+                    required="1"
```
Comment (bodedra) [inline scrap_reason_code/views/stock_scrap_views.xml]:
```suggestion
```

**Review by bodedra — CHANGES_REQUESTED** [review]
This PR will force to provide reason code for scrap order.

Reason code is not required for all inventory types. Intent of this module is to give flexibility to decide what kind of inventory required reason code or not.

I am good with multi company support changes.

File: scrap_reason_code/views/stock_scrap_views.xml [hunk 5]
```diff
@@ -17,7 +21,11 @@
         <field name="inherit_id" ref="stock.stock_scrap_tree_view" />
         <field name="arch" type="xml">
             <xpath expr="//field[@name='product_uom_id']" position='after'>
-                <field name="reason_code_id" />
+                <field
+                    name="reason_code_id"
+                    required="1"
+                    domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
```
Comment (LoisRForgeFlow) [inline scrap_reason_code/views/stock_scrap_views.xml:25]: Don't do this here, add multi company security rules for the reason code model.

Comment (bizzappdev) [reply scrap_reason_code/views/stock_scrap_views.xml:25]: correct me if I am wrong. In V13 you can have multiple companies active at the same time. so reason can not be from the current company of user but it has to be the company from the stock scrap/Stock picking is in

File: scrap_reason_code/views/stock_scrap_views.xml (hunk 4 above)
Comment (bizzappdev) [reply scrap_reason_code/views/stock_scrap_views.xml]: solved

File: scrap_reason_code/views/stock_scrap_views.xml (hunk 3 above)
Comment (bizzappdev) [reply scrap_reason_code/views/stock_scrap_views.xml]: sovled

File: scrap_reason_code/views/stock_scrap_views.xml (hunk 2 above)
Comment (bizzappdev) [reply scrap_reason_code/views/stock_scrap_views.xml]: sovled

File: scrap_reason_code/views/stock_scrap_views.xml (hunk 5 above)
Comment (LoisRForgeFlow) [reply scrap_reason_code/views/stock_scrap_views.xml:25]: Same as in the other PR, you are right, and you can keep it, but multicompany rules can still be added (and I think they should) and even a constraint to check that no inconsistency is entered.

Comment (bizzappdev) [reply scrap_reason_code/views/stock_scrap_views.xml:25]: Done

File: scrap_reason_code/views/stock_scrap_views.xml (hunk 1 above)
Comment (bizzappdev) [reply scrap_reason_code/views/stock_scrap_views.xml:12]: record rules are also added.

**Comment by dreispt** [comment]
There are changes to be re-reviewed. Thanks!

**Comment by bodedra** [comment]
I have tested this with multi company scenario and looks little ugly.

This allow me to pick my other allowed company to create scrap reason code.

Screenshot from 2021-02-16 12-09-07

But when I save, it throws multi company record rules validation. 

Screenshot from 2021-02-16 12-09-01

IMO it should only display currently logged in company in drop down Company field. Or any other suggestion?

**Comment by bizzappdev** [comment]
@bodedra It is exactly the same behaviour as the product model or any other model which has company m2o is there.
Selection_068
Selection_069

**Review by dreispt — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@bodedra @LoisRForgeFlow Could you update your review ?

File: scrap_reason_code/security/scrap_reason_code_security.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <data noupdate="0">
```
Comment (bodedra) [inline scrap_reason_code/security/scrap_reason_code_security.xml:3]:
```suggestion
    <data noupdate="1">
```

**Review by bodedra — CHANGES_REQUESTED** [review]
minor

**Comment by rousseldenis** [comment]
@bizzappdev Can you change noupdate entry ? Thanks

**Comment by rousseldenis** [comment]
@bizzappdev

**Comment by rousseldenis** [comment]
@bizzappdev

**Pull request closed without merging**